src/coreComponents/physicsSolvers/fluidFlow/SourceFluxStatistics.cpp

```diff
@@ -101,84 +128,88 @@ void SourceFluxStatsAggregator::registerDataOnMesh( Group & meshBodies )
-                               wrappedStats.stats().m_productionRate[0], massUnit ) );
+      tableData.addRow( elementSetName,
+                        GEOS_FMT( "{}", wrappedStats.stats().m_producedMass[0] ),
+                        GEOS_FMT( "{}", wrappedStats.stats().m_productionRate[0] ) );
     }
     else
     {
-      GEOS_LOG_RANK( GEOS_FMT( "{} {} (of {}, in {}): Produced mass = {} {}",
-                               catalogName(), getName(), wrappedStats.getFluxName(), elementSetName,
-                               wrappedStats.stats().m_producedMass, massUnit ) );
-      GEOS_LOG_RANK( GEOS_FMT( "{} {} (of {}, in {}): Production rate = {} {}/s",
-                               catalogName(), getName(), wrappedStats.getFluxName(), elementSetName,
-                               wrappedStats.stats().m_productionRate, massUnit ) );
+      tableData.addRow( elementSetName,
+                        GEOS_FMT( "{}", wrappedStats.stats().m_producedMass ),
+                        GEOS_FMT( "{}", wrappedStats.stats().m_productionRate ) );
     }
   }
 }
 
-void SourceFluxStatsAggregator::writeStatsToCSV( string_view elementSetName, WrappedStats const & stats,
-                                                 bool writeHeader )
+void SourceFluxStatsAggregator::outputStatsToLog( bool logLevelActive, string_view elementSetName, TableData const & tableMeshData )
 {
-  if( m_writeCSV > 0 && MpiWrapper::commRank() == 0 )
+  if( logLevelActive && logger::internal::rank == 0 )
   {
-    string const fileName = GEOS_FMT( "{}/{}_{}_{}.csv",
-                                      m_outputDir,
-                                      stats.getAggregatorName(), stats.getFluxName(), elementSetName );
-    std::ofstream outputFile( fileName,
-                              writeHeader ? std::ios_base::out : std::ios_base::app );
-    if( writeHeader )
-    {
-      outputFile << GEOS_FMT( "Time [s],Element Count,Producted Mass [{0}],Production Rate [{0}/s]",
-                              units::getSymbol( m_solver->getMassUnit() ) ) << std::endl;
-    }
-    else
+    m_logLayout.setTitle( GEOS_FMT( "Source flux statistics in {}", elementSetName ));
+    TableTextFormatter const tableStatFormatter( m_logLayout );
+    GEOS_LOG_RANK( tableStatFormatter.toString( tableMeshData ) );
+  }
+}
+void SourceFluxStatsAggregator::outputStatsToCSV( string_array const & filenames, TableData & csvData )
+{
+  if( m_writeCSV > 0 && logger::internal::rank == 0 )
+  {
+    for( auto const & filename : filenames )
     {
-      outputFile << GEOS_FMT( "{},{},{},{}",
-                              stats.getStatsPeriodStart(), stats.stats().m_elementCount,
-                              stats.stats().m_producedMass, stats.stats().m_productionRate ) << std::endl;
+      std::ofstream outputFile( filename );
+
+      TableCSVFormatter const tableStatFormatter( m_csvLayout );
+      outputFile << tableStatFormatter.toString( csvData );
```

Comment:
is it still going to be normal CSV?

Reply:
Yes column nale and values, each, appended with ","